services/price-service.js

```diff
@@ -3,12 +3,16 @@ const CouponData = require('../data/coupon-data')
 const DiscountDomain = require('../domains/discount-domain')
 const ItemDomain = require('../domains/item-domain')
 const OutOfStockError = require('../errors/out-of-stock-error')
+const ItemNotSupportCoupon = require('../errors/item-not-support-coupon')
 
 async function getPrice (priceRequest) {
   const item = await ItemData.getItemByCode(priceRequest.itemcode)
   const couponCode = priceRequest.coupon
   const coupon = couponCode ? await CouponData.getCouponByCode(couponCode) : null
 
+  if (ItemDomain.thisItemNotSupportCoupon(item, coupon)) {
+    throw new ItemNotSupportCoupon('Item not support coupon')
```

Comment:
As our old implementation when coupon is not valid we return same price with message error, so did should be use same heuristic for consistency.